### PR TITLE
:bug: Fix issue between homebrew/sudo and ansible become

### LINKED ISSUE #181: Vault of sudo password not working anymore
Since 2-3 days I have this when I execute the playbook:

CleanShot 2023-03-15 at 14 20 56

I try to re-execute `generate-ansible-password` without success.

### PR DESCRIPTION
**Pull request by notFloran**
Fix #181

Related to #179 : the fix I made broke the password pass between ansible and sudo.

`dev dotfiles` must be run to set `HOMEBREW_CACHE` export

### CODE REVIEW

File: default.config.yml [hunk 1]
```diff
@@ -4,8 +4,7 @@
 # ansible  #
 ############
 
-ansible_become_method: sudo 
-ansible_become_flags: "-E"
+ansible_become_flags: "-H -S -n --preserve-env=HOMEBREW_CACHE"
```
Comment (notFloran) [inline default.config.yml:7]: `-E` pass all env variables, is a security issue.

Here, we pass only the required env variable.

`-H -S -n` is the default value of ansible, `-S` is for the password

File: roles/dotfiles/templates/zshrc.j2 [hunk 2]
```diff
@@ -16,6 +16,9 @@ DEFAULT_USER="{{ lookup('env','USER') }}"
 
 export PATH=$HOME/bin:/usr/local/bin:/usr/local/sbin:$PATH
 
+# See https://github.com/Homebrew/brew/blob/master/Library/Homebrew/brew.sh#L48 for default value
+export HOMEBREW_CACHE={{ homebrew_cache_path }}
```
Comment (notFloran) [inline roles/dotfiles/templates/zshrc.j2:20]: homebrew with sudo need to know where is the cache

**Comment by nclsHart** [comment]
I've got an issue which seems to be related to brew run as sudo (during `dev setup tags=nginx`).

```
fatal: [127.0.0.1]: FAILED! => changed=true
  cmd:
  - brew
  - services
  - restart
  - nginx
  delta: '0:00:00.830175'
  end: '2023-04-25 14:06:08.832309'
  msg: non-zero return code
  rc: 1
  start: '2023-04-25 14:06:08.002134'
  stderr: 'Error: Need to download https://formulae.brew.sh/api/formula.jws.json but cannot as root! Run `brew update` without `sudo` first then try again.'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
```

I tried to run `brew update` and run again the setup command but without success.

**Comment by notFloran** [comment]
> I've got an issue which seems to be related to brew run as sudo (during `dev setup tags=nginx`).
> 
> ```
> fatal: [127.0.0.1]: FAILED! => changed=true
>   cmd:
>   - brew
>   - services
>   - restart
>   - nginx
>   delta: '0:00:00.830175'
>   end: '2023-04-25 14:06:08.832309'
>   msg: non-zero return code
>   rc: 1
>   start: '2023-04-25 14:06:08.002134'
>   stderr: 'Error: Need to download https://formulae.brew.sh/api/formula.jws.json but cannot as root! Run `brew update` without `sudo` first then try again.'
>   stderr_lines: <omitted>
>   stdout: ''
>   stdout_lines: <omitted>
> ```
> 
> I tried to run `brew update` and run again the setup command but without success.

Can you run `dev dotiles` ?

**Comment by nclsHart** [comment]
> Can you run dev dotiles ?

Done. But no changes :/

**Comment by nclsHart** [comment]
It works... I just forget to reload my zsh config.

**Pull request merged**